### PR TITLE
Gh 1980 fix parameterized loader its

### PR DESCRIPTION
**Pull request by m55624**

### CODE REVIEW

File: core/exception/src/main/java/uk/gov/gchq/gaffer/core/exception/Status.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public static Status fromStatusCode(final int statusCode) {
                 .stream()
                 .filter(v -> v.getStatusCode() == statusCode)
                 .findFirst()
-                .orElseThrow(RuntimeException::new);
+                .orElseThrow(() -> new GafferRuntimeException("Status code not recognised: " + statusCode));
```
Comment (GCHQDev404) [inline core/exception/src/main/java/uk/gov/gchq/gaffer/core/exception/Status.java]: 👍

File: core/store/src/main/java/uk/gov/gchq/gaffer/store/TypeReferenceStoreImpl.java [hunk 2]
```diff
@@ -30,6 +32,9 @@
     public static class StoreTraits extends TypeReference<Set<StoreTrait>> {
     }
 
+    public static class Operations extends TypeReference<Set<Class<? extends Operation>>> {
```
Comment (GCHQDev404) [inline core/store/src/main/java/uk/gov/gchq/gaffer/store/TypeReferenceStoreImpl.java]: neater

File: integration-test/src/test/java/uk/gov/gchq/gaffer/integration/impl/NoAggregationIT.java [hunk 3]
```diff
@@ -129,6 +129,7 @@ protected Schema createSchema() {
                 .edge(TestGroups.EDGE,
                         new SchemaEdgeDefinition.Builder()
                                 .source(TestTypes.ID_STRING)
+                                .directed(TestTypes.DIRECTED_EITHER)
```
Comment (GCHQDev404) [inline integration-test/src/test/java/uk/gov/gchq/gaffer/integration/impl/NoAggregationIT.java]: **To highlight this line**
This type definition has no validation function.
So directed can be either True or False.

File: store-implementation/proxy-store/src/main/java/uk/gov/gchq/gaffer/proxystore/ProxyStore.java [hunk 4]
```diff
@@ -102,8 +102,7 @@ protected void checkDelegateStoreStatus() throws StoreException {
     @SuppressFBWarnings(value = "SIC_INNER_SHOULD_BE_STATIC_ANON")
     protected Set<Class<? extends Operation>> fetchOperations() throws StoreException {
         final URL url = getProperties().getGafferUrl("graph/operations");
-        return (Set) Collections.unmodifiableSet(doGet(url, new TypeReference<Set<Class<Operation>>>() {
-        }, null));
+        return Collections.unmodifiableSet(doGet(url, new TypeReferenceStoreImpl.Operations(), null));
```
Comment (GCHQDev404) [inline store-implementation/proxy-store/src/main/java/uk/gov/gchq/gaffer/proxystore/ProxyStore.java]: tidy.

**Comment by m55624** [comment]
This is now at a stage of consistently working, apart from ParquetSchemaHidingIT which fails due to aggregation, which is a known bug.  A skip for these tests needed then should be good to merge.

**Pull request merged**